sections/Deals/CreateDeal.tsx

```diff
@@ -264,6 +264,10 @@ const CreateDeal: FC<CreateDealProps> = ({ poolAddress, purchaseToken }) => {
 		};
 	}, [formik?.values, poolAddress, provider, signer, walletAddress]);
 
+	const underlyingDealTokenTotal = useMemo(() => {
+		return Number(formik.values.dealExchangeRate) * Number(formik.values?.purchaseTokenTotal ?? 0);
```

Comment:
because of javascript math issues, let's use wei here.

Reply:
maybe something like this `wei(formik.values.dealExchangeRate, 0).mul(formik.values?.purchaseTokenTotal ?? 0).toNumber();`

Reply:
@0xcdb we need to all go over the wei library in detail at some point. I think something like this will work. can you confirm?